script/deploy/mainnet/v0.3.0-mainnet-rewards.s.sol

```diff
@@ -0,0 +1,100 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.12;
+
+import "@openzeppelin/contracts/utils/Create2.sol";
+import "../../utils/ExistingDeploymentParser.sol";
+
+/**
+ * @notice Script used for the first deployment of EigenLayer core contracts to Holesky
+ * FORK LOCAL
+ * anvil --fork-url $RPC_MAINNET
+ * forge script script/deploy/mainnet/v0.3.0-mainnet-rewards.s.sol:MainnetRewardsCoordinatorDeploy --rpc-url http://127.0.0.1:8545 --private-key $PRIVATE_KEY --broadcast -vvvv
+ *
+ * MAINNET
+ * forge script script/deploy/mainnet/v0.3.0-mainnet-rewards.s.sol:MainnetRewardsCoordinatorDeploy --rpc-url $RPC_MAINNET --private-key $PRIVATE_KEY --verify --broadcast -vvvv
+ *
+ */
+contract MainnetRewardsCoordinatorDeploy is ExistingDeploymentParser {
+    function run() external virtual {
+        _parseInitialDeploymentParams(
+            "script/configs/mainnet/v0.3.0-mainnet-rewards.config.json"
+        );
+        _parseDeployedContracts(
+            "script/configs/mainnet/v0.3.0-eigenlayer-addresses.config.json"
+        );
+
+        // START RECORDING TRANSACTIONS FOR DEPLOYMENT
+        vm.startBroadcast();
+
+        emit log_named_address("Deployer Address", msg.sender);
+
+        _deployRewardsCoordinator();
+
+        // STOP RECORDING TRANSACTIONS FOR DEPLOYMENT
+        vm.stopBroadcast();
+
+        // Sanity Checks
+        _verifyContractPointers();
+        _verifyImplementations();
+        _verifyContractsInitialized({isInitialDeployment: true});
+        _verifyInitializationParams();
```

Comment:
so these checks aren't actually relevant to the new deployment?
would recommend replacing these with something more relevant, or at least adding some RewardsCoordinator-specific automated checks

Reply:
The rewardsCoordinator params in the json `script/configs/mainnet/v0.3.0-mainnet-rewards.config.json` are checked in the ExistingDeploymentParser helper. Specifically the `_verifyInitializationParams` function call right here https://github.com/Layr-Labs/eigenlayer-contracts/blob/rewards-mainnet-deploy/script/utils/ExistingDeploymentParser.sol#L554-L593
Personally would like to keep the other functions in as additional sanity checks to ensure the correct addresses used (since we have different config files for holesky and mainnet)

Reply:
oh, neat! thanks, that's my bad; resolving